src/ng/parse.js

```diff
@@ -1788,6 +1797,7 @@ function $ParseProvider() {
             }
             cache[cacheKey] = parsedExpression;
           }
+          parsedExpression = expensiveChecksInterceptor(parsedExpression, expensiveChecks);
```

Comment:
> Also, this would hide properties like assign, constant, inputs and literal

Given that the expensive checks is a private API, I think this is not an issue. Anyhow, made the change.

> Couldn't we move this inside the if-block ?

Yep :)